monai/handlers/mlflow_handler.py

```diff
@@ -352,3 +402,46 @@ def _default_iteration_log(self, engine: Engine) -> None:
                     for i, param_group in enumerate(cur_optimizer.param_groups)
                 }
                 self._log_metrics(params, step=engine.state.iteration)
+
+    def _default_dataset_log(self, engine: Engine) -> None:
+        """
+        Execute dataset log operation based on MONAI `Workflow.data_loader.dataset` data.
+        Abstract sample names in a dataset and build a Pandas DataFrame from it. To use this
+        function, every sample in the input dataset must have a filename, which can be fetched
+        from the `filename_or_obj` parameter in the `image_meta_dict` of the sample.
+        This function will log a PandasDataset, generated from the Pandas DataFrame, to MLFlow
+        inputs. For more details about PandasDataset, please refer to this link:
+        https://mlflow.org/docs/latest/python_api/mlflow.data.html#mlflow.data.pandas_dataset.PandasDataset
+
+        Please note that it may take a while to record the dataset if it has too many samples.
+
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+
+        """
+        dataloader = getattr(engine, "data_loader", None)
+        dataset = getattr(dataloader, "dataset", None) if dataloader else None
```

Comment:
Can we pass the dataset object to `self.log_dataset` during initialization?
if dataset is not None, then use it to log the information.
Similar to: https://github.com/Project-MONAI/MONAI/blob/dev/monai/handlers/lr_schedule_handler.py#L39.
What do you think?

Thanks.